grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java

```diff
@@ -61,12 +80,19 @@ public void onFailure(Call<Repositories> call, Throwable t) {
     }
 
     private void showRepositories() {
-        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_main_activity);
-        mRecyclerView.setHasFixedSize(true);
-        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
-
-        mAdapter = new RepositoryAdpter(mRepositoryItems);
-
-        mRecyclerView.setAdapter(mAdapter);
+        if (mAdapter == null) {
+            mAdapter = new RepositoryAdpter(mRepositoryItems, mRecyclerView);
+            mAdapter.setOnLoadMoreListener(new RepositoryAdpter.OnLoadMoreListener() {
+                @Override
+                public void onLoadMore() {
+                    fetchRepositories();
+                }
+            });
+
+            mRecyclerView.setAdapter(mAdapter);
+            return;
+        }
+
+        mAdapter.notifyDataSetChanged();
```

Comment:
to avoid null check I would of wrote the class like time. 

``` java
public class MainActivity extends AppCompatActivity {

    private final String TAG = MainActivity.class.getSimpleName();

    private final int LOAD_MORE_THRESHOLD = 2;

    private ProgressBar mProgress;
    private LinearLayoutManager mLinearLayoutManager;
    private RepositoryAdpter mAdapter;

    private int mPage = 0;
    private int total = 0;
    private int last = 0;
    private boolean isLoading = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mProgress = (ProgressBar) findViewById(R.id.progress_main_activity);
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recycler_main_activity);

        mLinearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(mLinearLayoutManager);
        recyclerView.setHasFixedSize(true);
        recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
            @Override
            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
                total = mLinearLayoutManager.getItemCount();
                last = mLinearLayoutManager.findLastVisibleItemPosition();

                if (dy < 0) {
                    return;
                }

                if (!isLoading && last + LOAD_MORE_THRESHOLD >= total) {
                    fetchRepositories();
                    mAdapter.setIsLoading(true);
                    isLoading = true;
                }
            }
        });

        mAdapter = new RepositoryAdpter();
        recyclerView.setAdapter(mAdapter);

        fetchRepositories();
    }

    private void fetchRepositories() {
        GithubApi api = ApiClient.getClient().create(GithubApi.class);

        Call<Repositories> call = api.getRepositories("language:java", "Repositories", ++mPage + "", "100");
        call.enqueue(new Callback<Repositories>() {
            @Override
            public void onResponse(Call<Repositories> call, Response<Repositories> response) {
                Log.d(TAG, "success");
                mProgress.setVisibility(View.GONE);

                mAdapter.add(response.body().items);
                mAdapter.setIsLoading(false);
                isLoading = false;
            }

            @Override
            public void onFailure(Call<Repositories> call, Throwable t) {
                Log.e(TAG, "failure");
                mProgress.setVisibility(View.GONE);
            }
        });
    }
}
```